docs/tremor-query/index.md

```diff
@@ -120,6 +120,13 @@ Tremor supports tumbling windows by number of events or by time.
 General configuration Parameters:
 
 * `eviction_period`: duration in nanoseconds without events arriving, after which to evict / remove the current window data for a single group.
+* `max_groups`: maximum number of groups to maintain simultaneously in memory. Groups added beyond that number will be ignored. Per default, tremor allows as much groups as will fit into memory.
+
+Each select statement maintains the groups for the current windows in an in memory data-structure. This contains the group values as well as the aggregate states.
+If your grouping values possibly have a very high cardinality it is possible to end up with runaway memory growth, as per default the group data structures won't be evicted,
+unless `eviction_period` is set. Old groups will be discarded after `2 x eviction_period` if no event for those groups arrived.
```

Comment:
Is this correct? I was under the impression that they get evicted as well when they emit unless `emit_empty_windows` is selected.

Reply:
when they emit, the aggregate states get reset, but the groups remain in the internal map inside the window.

I think we might need to spend some time coming up with a better default behaviour that cleans up as quickly as possible (in an LRU manner maybe).

Reply:
Ja this sounds totally wrong to me IMHO, let's make a ticket and clean that up, I see horrible failure cases in here :joy: (but not related to the doc PR so a nonissue here)